website/addons/s3/static/s3NodeConfig.js

```diff
@@ -0,0 +1,360 @@
+'use strict';
+var ko = require('knockout');
+require('knockout-punches');
+require('knockout-mapping');
+var $ = require('jquery');
+var bootbox = require('bootbox');
+var Raven = require('raven-js');
+
+var $osf = require('js/osfHelpers');
+
+ko.punches.enableAll();
+
+var noop = function() {};
+
+var ViewModel = function(url, selector) {
+    var self = this;
+
+    self.url = url;
+    self.selector = selector;
+
+    self.nodeHasAuth = ko.observable(false);
+    self.userHasAuth = ko.observable(false);
+    self.userIsOwner = ko.observable(false);
+    self.ownerName = ko.observable('');
+
+    self.urls = ko.observable({});
+    self.loadedSettings = ko.observable(false);
+    self.bucketList = ko.observableArray([]);
+    self.loadedBucketList = ko.observable(false);
+    self.currentBucket = ko.observable('');
+    self.selectedBucket = ko.observable('');
+
+    self.accessKey = ko.observable('');
+    self.secretKey = ko.observable('');
+
+    self.loading = ko.observable(false);
+    self.creating = ko.observable(false);
+    self.creatingCredentials = ko.observable(false);
+
+    self.message = ko.observable('');
+    self.messageClass = ko.observable('text-info');
+
+    self.showSelect = ko.observable(false);
+
+    self.showSettings = ko.pureComputed(function() {
+        return self.nodeHasAuth();
+    });
+    self.disableSettings = ko.pureComputed(function() {
+        return !(self.userHasAuth() && self.userIsOwner());
+    });
+    self.showNewBucket = ko.pureComputed(function() {
+        return self.userHasAuth() && self.userIsOwner();
+    });
+    self.showImport = ko.pureComputed(function() {
+        return self.userHasAuth() && !self.nodeHasAuth();
+    });
+    self.showCreateCredentials = ko.pureComputed(function() {
+        return !self.nodeHasAuth() && !self.userHasAuth();
+    });
+    self.canChange = ko.pureComputed(function() {
+        return self.userIsOwner() && self.nodeHasAuth();
+    });
+    self.allowSelectBucket = ko.pureComputed(function() {
+        return (self.bucketList().length > 0 || self.loadedBucketList()) && (!self.loading());
+    });
+};
+
+ViewModel.prototype.toggleSelect = function() {
+    var self = this;
+    self.showSelect(!self.showSelect());
+    return self.fetchBucketList()
+    .done(function(buckets){
+        self.bucketList(buckets);
+        self.selectedBucket(self.currentBucket());
+    });
+};
+
+ViewModel.prototype.selectBucket = function() {
+    var self = this;
+    self.loading(true);
+    return $osf.postJSON(
+            self.urls().set_bucket, {
+                's3_bucket': self.selectedBucket()
+            }
+        )
+        .done(function(response) {
+            self.updateFromData(response);
+            var filesUrl = window.contextVars.node.urls.web + 'files/';
```

Comment:
`contextVars` should not be used in reusable modules. Anything that needs to be pulled from the global context should be passed to the class as an argument.